### PR TITLE
Feature/user session

### PR DESCRIPTION
**Pull request by shogunpurple**
## Description
Just raising PR for the work @mike12345567 has done around session management.

We were using a fairly rudimentary default JWT session management mechanism before in budibase. This came with a few issues:
- Sessions were only timed from login. This meant you could be using the product the entire time, but when your expiry date came, you would be logged out. Your TTL should contantly be extended to keep your session alive during live use.
- Sessions information was entirely stored on the JWT issued to the client. Having this information in the cache makes it much easier to debug, query and interact with session data.

This PR contains a complete solution that moves all budibase sessions to redis and provides ways to interact with them at all levels of the stack.

#### Session Management API and Management Clients
This PR contains client libraries that wrap redis for complete session management - creation of new sessions, invalidation, fetching current sessions and TTL extension.

These client libraries can be called directly or through a new HTTP API for session management. The main plan here for this HTTP API is to implement a session management UI accessible by admins: https://github.com/Budibase/budibase/issues/1949

#### Updates to JWT
Added the sessionID to the cookie so we know which session we are reading from redis on each request.

### User cache
User information is now cached in redis to avoid additional reads every time we need user information. In general, this happens a lot because we fetch this information on every request through the `authenticated` middleware. Moving this to redis takes a load off CouchDB and the server in general, and should drop latency on response times!

### Notes
- Default session expiry is 24 hours, but will be extended through use. Your session will only expire when it hasn't been touched for some time.

### CODE REVIEW

**Review by aptkingston — APPROVED** [review]
LGTM! Had discussed this with Mike before and this makes far more sense for managing user sessions than storing data in the cookie which immediately goes stale (and can't easily be revoked or changed). Love the inclusion of the admin API too 💯

File: packages/worker/src/api/controllers/admin/auth.js [hunk 1]
```diff
@@ -10,7 +10,7 @@ const { checkResetPasswordCode } = require("../../../utilities/redis")
 
 const GLOBAL_DB = authPkg.StaticDatabases.GLOBAL.name
 
-function authInternal(ctx, user, err = null) {
+async function authInternal(ctx, user, err = null) {
```
Comment (Rory-Powell) [inline packages/worker/src/api/controllers/admin/auth.js:13]: How come we made this one async?

Comment (mike12345567) [reply packages/worker/src/api/controllers/admin/auth.js:13]: Good catch @Rory-Powell - I had planned to do some async work in this function but ended up moving it further up the chain, forgot to put this back to sync, will fix on my next branch!

Comment (Rory-Powell) [reply packages/worker/src/api/controllers/admin/auth.js:13]: ah okay that makes sense, sounds good

**Pull request merged**